scripts/create_registry_schema.py

```diff
@@ -59,39 +60,103 @@ def _get_column_definitions(schema, table):
-        if schema_yaml[table][column]["foreign_key"]:
+        if schema_data[table]["column_definitions"][column]["foreign_key"]:
             fk_schema = schema
-            if schema_yaml[table][column]["foreign_key_schema"] != "self":
-                fk_schema = schema_yaml[table][column]["foreign_key_schema"]
+            if (
+                schema_data[table]["column_definitions"][column]["foreign_key_schema"]
+                != "self"
+            ):
+                fk_schema = schema_data[table]["column_definitions"][column][
+                    "foreign_key_schema"
+                ]
 
             return_dict[column] = Column(
                 column,
-                _TYPE_TRANSLATE[schema_yaml[table][column]["type"]],
+                _TYPE_TRANSLATE[
+                    schema_data[table]["column_definitions"][column]["type"]
+                ],
                 ForeignKey(
                     _get_ForeignKey_str(
                         fk_schema,
-                        schema_yaml[table][column]["foreign_key_table"],
-                        schema_yaml[table][column]["foreign_key_column"],
+                        schema_data[table]["column_definitions"][column][
+                            "foreign_key_table"
+                        ],
+                        schema_data[table]["column_definitions"][column][
+                            "foreign_key_column"
+                        ],
                     )
                 ),
-                primary_key=schema_yaml[table][column]["primary_key"],
-                nullable=schema_yaml[table][column]["nullable"],
+                primary_key=schema_data[table]["column_definitions"][column][
+                    "primary_key"
+                ],
+                nullable=schema_data[table]["column_definitions"][column]["nullable"],
             )
 
         # Normal case
         else:
             return_dict[column] = Column(
                 column,
-                _TYPE_TRANSLATE[schema_yaml[table][column]["type"]],
-                primary_key=schema_yaml[table][column]["primary_key"],
-                nullable=schema_yaml[table][column]["nullable"],
+                _TYPE_TRANSLATE[
+                    schema_data[table]["column_definitions"][column]["type"]
+                ],
+                primary_key=schema_data[table]["column_definitions"][column][
+                    "primary_key"
+                ],
+                nullable=schema_data[table]["column_definitions"][column]["nullable"],
             )
 
     return return_dict
 
 
+def _get_table_metadata(schema, table):
+    """
+    Build the table meta data dict, e.g., the schema name and any unique
+    constraints, for this table.
+
+    Parameters
+    ----------
+    schema : str
+    table : str
+
+    Returns
+    -------
+    meta : dict
+        The table metadata
+    """
+
+    # Table metadata
+    meta = {
+        "__tablename__": table,
+    }
+
+    if (
```

Comment:
This needs to be done in a more general fashion; there are several cases it doesn't handle:

- multiple uniqueness constraints
- multiple indices
- one or more indices but no uniqueness constraints.

The first thing to do is to come up with a better representation of these things in the schema yaml file which supports multiple indices and constraints in a single table. We don't have any need for that now but someday we might. Write a routine which can handle a list of uniqueness constraints, outputting the sort of metadata sqlalchemy expects, and another for a list of indices. 

Then, rather than having `if` ... `elif` ... `elif`...  to handle all the cases, it would be somewhat nicer (but not crucial) to add on metadata for uniqueness constraints if there are any, then the same thing for indices.

Reply:
Think I have this now, should be able to add an arbitrary number of unique constraints and indexes to each table.

I've also added a doc string to the schema.yaml file